crates/subspace-farmer/src/cluster/cache.rs

```diff
@@ -325,29 +332,24 @@ async fn send_identify_broadcast<C>(
 ) where
     C: PieceCache,
 {
-    caches_details
+    let Some(cache_id) = caches_details.first().map(|cache| cache.cache_id) else {
+        // No caches, nothing to do
+        return;
+    };
+    let details = caches_details
         .iter()
-        .map(|cache| async move {
-            if let Err(error) = nats_client
-                .broadcast(
-                    &ClusterCacheIdentifyBroadcast {
-                        cache_id: cache.cache_id,
-                        max_num_elements: cache.cache.max_num_elements(),
-                    },
-                    cache_group,
-                )
-                .await
-            {
-                warn!(
-                    cache_id = %cache.cache_id,
-                    %error,
-                    "Failed to send cache identify notification"
-                );
-            }
+        .map(|cache| ClusterCacheIdentifyDetail {
+            cache_id: cache.cache_id,
+            max_num_elements: cache.cache.max_num_elements(),
         })
-        .collect::<FuturesUnordered<_>>()
-        .collect::<Vec<_>>()
-        .await;
+        .collect::<Vec<_>>();
+
+    if let Err(error) = nats_client
+        .broadcast(&ClusterCacheIdentifyBroadcast { details }, cache_group)
+        .await
+    {
+        warn!(%cache_id, %error, "Failed to send cache identify notification");
```

Comment:
Optional: Previously, we would send a log for every failed `cache_id`, now we're only logging the first one. If that's what we want to do, can we make it clearer in the code?

Here are some ideas:

Rename the variable (and above where the variable is initialised):
```rust
        warn!(%first_cache_id, %error, "Failed to send cache identify notification");
```

Or log the number of caches (and a comment):
```rust
        // We only log the first cache_id in the broadcast
        let cache_count = caches_details.len();
        warn!(%cache_id, %cache_count, %error, "Failed to send cache identify notification");
```

Reply:
log the number is a good idea for me, let me fix it